### ISSUE TITLE
Question about Transfer from Litbank to Timebank

### ISSUE THREAD

**Issue by xixy** [aakanksha19/ODETTE#2]
Hi, can we just use the current code for transfer from litbank to timebank? My concern is the proportion of source/target data in adv_batches. With timebank/litbank as source/target domain, the proportion is 1:1. However, train set of litbank has 5263 sentences, much more than train set of timebank which has 1531 sentences. The following setting doesn't guarantee a 1:1 proportion

adv_sents = unlabeled_sents[:len(labeled_sents)] + labeled_sents
adv_parse = unlabeled_parse[:len(labeled_sents)] + labeled_parse
adv_domains = unlabeled_domains[:len(labeled_sents)] + labeled_domains

Thanks!

**Comment by aakanksha19** [comment]
Hi! Yes, you can use the current code as is for transfer from litbank to timebank to replicate the paper results. We did not balance the source/target for litbank to timebank transfer because we were worried that if we undersampled litbank in this setting (to 1/4th its size), we would end up with many words from the literature domain that appear in the labeled training data but not in the unlabeled data. In this scenario, we were worried that embeddings for those words would be updated by the event classifier but never by the adversarial setup, leading the classifier to have some domain-specific information associated with these missing words.  This is not an issue in timebank->litbank transfer, because timebank is the labeled source data and is not being undersampled. If you would like to balance the data for litbank to timebank, you could try oversampling timebank instead (adding token dropout during oversampling). Additionally, if you are using BERT models, missing vocabulary is not an issue, so undersampling litbank may be okay. Hope this helps!
